x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1342,6 +1421,23 @@ private static Expression castStringLiteral(Expression from, DataType target) {
         }
     }
 
+    private static class ImplicitForkCasting extends ParameterizedRule<LogicalPlan, LogicalPlan, AnalyzerContext> {
+        private final ImplicitCasting implicitCasting = new ImplicitCasting();
```

Comment:
I do wonder if there's a way to get this for free by modifying the other rules. I'm worried that we'll forget one and it won't work in forks when it should. But that's more of a @costin question.

Reply:
I deliberately avoided generalising this for now, since Fork is still snapshot only, and changes are likely to be more evasive. However, we can do so now, if you have a concrete suggestion of how best to do it. Otherwise, let's do it as a follow up.

Reply:
oh yeah, a follow up is fine with me.

Reply:
cool.